airbyte-integrations/connectors/source-mssql/src/test-integration/java/io/airbyte/integrations/source/mssql/MssqlSourceAcceptanceTest.java

```diff
@@ -139,6 +158,79 @@ protected void testAddNewStreamToExistingSync() throws Exception {
     assertEquals(SCHEMA_NAME, stateMessages2.get(1).getStream().getStreamDescriptor().getNamespace());
   }
 
+  @Test
+  protected void testNullValueConversion() throws Exception {
+    final String STREAM_NAME3 = "stream3";
+    testdb.getDatabase().query(c -> {
+      return c.query("""
+                     CREATE TABLE %s.%s (id INTEGER PRIMARY KEY, name VARCHAR(200), userid INTEGER DEFAULT NULL);
+                     """.formatted(SCHEMA_NAME, STREAM_NAME3));
+    }).execute();
+
+    testdb.getDatabase().query(c -> {
+      return c.query("""
+                     INSERT INTO %s.%s (id, name) VALUES (4,'voyager');
+                     """.formatted(SCHEMA_NAME, STREAM_NAME3));
+    }).execute();
+
+    final List<ConfiguredAirbyteStream> configuredAirbyteStreams =
+        Lists.newArrayList(CatalogHelpers.createConfiguredAirbyteStream(STREAM_NAME3,
+            SCHEMA_NAME,
+            Field.of("id", JsonSchemaType.NUMBER),
+            Field.of("name", JsonSchemaType.STRING),
+            Field.of("userid", JsonSchemaType.NUMBER))
+            .withDestinationSyncMode(DestinationSyncMode.APPEND)
+            .withSyncMode(INCREMENTAL)
+            .withCursorField(List.of("id")));
+    final ConfiguredAirbyteCatalog configuredCatalogWithOneStream =
+        new ConfiguredAirbyteCatalog().withStreams(List.of(configuredAirbyteStreams.get(0)));
+
+    final List<AirbyteMessage> airbyteMessages = runRead(configuredCatalogWithOneStream, getState());
+    final List<AirbyteRecordMessage> recordMessages = filterRecords(airbyteMessages);
+    final List<AirbyteStateMessage> stateMessages = airbyteMessages
+        .stream()
+        .filter(m -> m.getType() == AirbyteMessage.Type.STATE)
+        .map(AirbyteMessage::getState)
+        .collect(Collectors.toList());
+    assertEquals(recordMessages.size(), 1);
+    assertFalse(stateMessages.isEmpty(), "Reason");
+    // TODO validate exact records
+    ObjectMapper mapper = new ObjectMapper();
+
+    assertTrue(recordMessages.get(0).getData().equals(mapper.readTree("""
+                                                                      {"id":4, "name":"voyager"}""")));
```

Comment:
nit: the formatter is to blame here but the text block doesn't benefit from being multiline

comment applies elsewhere also

Reply:
I removed all literal string blocks